org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/cleanup/CleanUpRegistry.java

```diff
@@ -75,10 +81,15 @@ public List<TextEdit> getEditsForAllActiveCleanUps(TextDocumentIdentifier textDo
 
 		IJavaProject javaProject = JDTUtils.resolveCompilationUnit(textDocumentId.getUri()).getJavaProject();
```

Comment:
In the code you add you do :

```
ICompilationUnit compilationUnit = JDTUtils.resolveCompilationUnit(documentUri);
if (compilationUnit != null) {
...
...
}
```

If you do that here also, you'll fix https://github.com/eclipse-jdtls/eclipse.jdt.ls/issues/2879 :+1: 

I think you can safely put :

```
if (compilationUnit == null) {
  return Collections.emptyList()
}
```

Reply:
Done with latest patch.